### PR TITLE
[SOAR-16875] - Palo Alto Pan OS - Addressing SNYK vulnerabilities in gunicorn and validators

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  -  Bumping requirements of `gunicorn` and `validators`
  -  update the SDK to 5.4.9 
  - Added examples to all actions 
  - Updated unit tests to include schema checks

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

testing actions via postman 

add a new policy
image

get_policy 
image

get logs 
image

### CODE REVIEW

File: plugins/palo_alto_pan_os/plugin.spec.yaml [hunk 1]
```diff
@@ -453,16 +516,19 @@ actions:
         description: Status of the requested operation e.g. success, error, etc
         type: string
         required: false
+        example: true
```
Comment (ablakley-r7) [inline plugins/palo_alto_pan_os/plugin.spec.yaml:519]:
```suggestion
        example: "success"
```
Not sure, maybe true is already a valid status here?

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]

File: plugins/palo_alto_pan_os/plugin.spec.yaml (hunk 1 above)
Comment (rbowden-r7) [reply plugins/palo_alto_pan_os/plugin.spec.yaml:519]: I have updated all of the status to "success"

as this is what we return on a successful request 

```
            "response": {
                "@code": "20",
                "@status": "success",
                "msg": "command succeeded"
            }
 ```

File: plugins/palo_alto_pan_os/komand_palo_alto_pan_os/util/util.py [hunk 2]
```diff
@@ -30,17 +30,17 @@ def extract_from_security_policy(self, policy: dict) -> dict:  # noqa: MC0001
             "hip-profiles",
         ]
         output = {}
-        for i in key_list:
+        for key in key_list:
             try:
-                output[i] = policy["response"]["result"]["entry"][i]["member"]
+                output[key] = policy["response"]["result"]["entry"][key]["member"]
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/palo_alto_pan_os/komand_palo_alto_pan_os/util/util.py:35]: it was left this as is, as the error handling would that was already there would address this case

```
            try:
                output[key] = policy["response"]["result"]["entry"][key]["member"]
            except KeyError:
                self.logger.info(f"Current policy {policy}")
                self.logger.info(f"The current policy has no {key} policy: Setting to any.")
                output[key] = "any"
            except TypeError:
                self.logger.info(f"Current policy {policy}")
                self.logger.info(f"The current policy has no policy config for {key}: Setting to any.")
                output[key] = "any"
            except BaseException:
                raise PluginException(
                    cause="An unknown formatting error occurred when formatting a security policy.",
                    assistance="Contact support for help.",
                    data=f"Policy config: {policy}",
                )
```

File: plugins/palo_alto_pan_os/komand_palo_alto_pan_os/connection/connection.py [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 
-        hostname = params.get("server")
-        verify_cert = params.get("verify_cert")
+        hostname = params.get(Input.SERVER)
+        verify_cert = params.get(Input.VERIFY_CERT)
 
-        username = params.get("credentials").get("username")
-        password = params.get("credentials").get("password")
+        username = params.get(Input.CREDENTIALS).get("username")
+        password = params.get(Input.CREDENTIALS).get("password")
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/palo_alto_pan_os/komand_palo_alto_pan_os/connection/connection.py:21]: The default has been updated here to {}

File: plugins/palo_alto_pan_os/komand_palo_alto_pan_os/connection/connection.py [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
         self.logger.info("Connect: Connecting..")
 
-        hostname = params.get("server")
-        verify_cert = params.get("verify_cert")
+        hostname = params.get(Input.SERVER)
+        verify_cert = params.get(Input.VERIFY_CERT)
 
-        username = params.get("credentials").get("username")
-        password = params.get("credentials").get("password")
+        username = params.get(Input.CREDENTIALS).get("username")
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/palo_alto_pan_os/komand_palo_alto_pan_os/connection/connection.py:20]: The default has been updated here to {}

File: plugins/palo_alto_pan_os/Dockerfile
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/palo_alto_pan_os/Dockerfile:1]: yes this should be able to be updated to nobody as there no specific code that requires that level of permission

**Review by ablakley-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

**Review by ekelly-r7 — APPROVED** [review]

**Pull request merged**